bk://kernel.bkbits.net/vojtech/input
vojtech@suse.cz|ChangeSet|20050113105509|25891 vojtech

# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/01/12 12:29:03-08:00 akpm@bix.(none) 
#   Merge bix.(none):/usr/src/bk25 into bix.(none):/usr/src/bk-input
# 
# Documentation/kernel-parameters.txt
#   2005/01/12 12:28:59-08:00 akpm@bix.(none) +0 -0
#   Auto merged
# 
# ChangeSet
#   2005/01/10 15:05:31-08:00 akpm@bix.(none) 
#   Merge bix.(none):/usr/src/bk25 into bix.(none):/usr/src/bk-input
# 
# drivers/usb/input/hid-core.c
#   2005/01/10 15:05:26-08:00 akpm@bix.(none) +0 -0
#   Auto merged
# 
# ChangeSet
#   2005/01/08 21:50:33-08:00 akpm@bix.(none) 
#   Merge bix.(none):/usr/src/bk25 into bix.(none):/usr/src/bk-input
# 
# Documentation/kernel-parameters.txt
#   2005/01/08 21:50:29-08:00 akpm@bix.(none) +0 -0
#   Auto merged
# 
# ChangeSet
#   2005/01/06 17:43:50+01:00 dtor_core@ameritech.net 
#   Input: psmouse - set mouse name to "Mouse" when using PS2++ and
#          don't have any other information about the mouse.
#   
#   Signed-off-by: Dmitry Torokhov <dtor@mail.ru>
#   Signed-off-by: Vojtech Pavlik <vojtech@suse.cz>
# 
# drivers/input/mouse/logips2pp.c
#   2005/01/06 17:43:44+01:00 dtor_core@ameritech.net +13 -2
#   Input: psmouse - set mouse name to "Mouse" when using PS2++ and
#          don't have any other information about the mouse.
#   
#   Signed-off-by: Dmitry Torokhov <dtor@mail.ru>
#   Signed-off-by: Vojtech Pavlik <vojtech@suse.cz>
# 
# ChangeSet
#   2005/01/06 17:42:15+01:00 dtor_core@ameritech.net 
#   Input: atkbd - fix keycode table size initialization that got broken
#          by my changes that exported 'set' and other settings via sysfs.
#          setkeycodes should work again now.
#   
#   Signed-off-by: Dmitry Torokhov <dtor@mail.ru>
#   Signed-off-by: Vojtech Pavlik <vojtech@suse.cz>
# 
# drivers/input/keyboard/atkbd.c
#   2005/01/06 17:42:09+01:00 dtor_core@ameritech.net +4 -4
#   Input: atkbd - fix keycode table size initialization that got broken
#          by my changes that exported 'set' and other settings via sysfs.
#          setkeycodes should work again now.
#   
#   Signed-off-by: Dmitry Torokhov <dtor@mail.ru>
#   Signed-off-by: Vojtech Pavlik <vojtech@suse.cz>
# 
# ChangeSet
#   2005/01/04 23:27:11-08:00 akpm@bix.(none) 
#   Merge bix.(none):/usr/src/bk25 into bix.(none):/usr/src/bk-input
# 
# Documentation/kernel-parameters.txt
#   2005/01/04 23:27:07-08:00 akpm@bix.(none) +0 -0
#   Auto merged
# 
# ChangeSet
#   2005/01/04 14:15:59-08:00 akpm@bix.(none) 
#   Merge bix.(none):/usr/src/bk25 into bix.(none):/usr/src/bk-input
# 
# Documentation/kernel-parameters.txt
#   2005/01/04 14:15:55-08:00 akpm@bix.(none) +0 -0
#   Auto merged
# 
# ChangeSet
#   2005/01/04 00:25:03-08:00 akpm@bix.(none) 
#   Merge bk://kernel.bkbits.net/vojtech/input
#   into bix.(none):/usr/src/bk-input
# 
# Documentation/kernel-parameters.txt
#   2005/01/04 00:24:59-08:00 akpm@bix.(none) +0 -0
#   Auto merged
# 
# ChangeSet
#   2005/01/04 00:23:10-08:00 akpm@bix.(none) 
#   Merge bix.(none):/usr/src/bk25 into bix.(none):/usr/src/bk-input
# 
# Documentation/kernel-parameters.txt
#   2005/01/04 00:23:05-08:00 akpm@bix.(none) +0 -0
#   Auto merged
# 
# ChangeSet
#   2004/12/29 18:24:03-08:00 akpm@bix.(none) 
#   Merge bix.(none):/usr/src/bk25 into bix.(none):/usr/src/bk-input
# 
# Documentation/kernel-parameters.txt
#   2004/12/29 18:23:57-08:00 akpm@bix.(none) +0 -0
#   Auto merged
# 
# ChangeSet
#   2004/12/28 20:48:26-08:00 akpm@bix.(none) 
#   Merge bk://kernel.bkbits.net/vojtech/input
#   into bix.(none):/usr/src/bk-input
# 
# drivers/usb/input/hid-core.c
#   2004/12/28 20:48:22-08:00 akpm@bix.(none) +0 -0
#   Auto merged
# 
# drivers/input/tsdev.c
#   2004/12/28 20:48:22-08:00 akpm@bix.(none) +0 -0
#   Auto merged
# 
# drivers/input/mousedev.c
#   2004/12/28 20:48:22-08:00 akpm@bix.(none) +0 -0
#   Auto merged
# 
# drivers/input/joydev.c
#   2004/12/28 20:48:22-08:00 akpm@bix.(none) +0 -0
#   Auto merged
# 
# drivers/input/evdev.c
#   2004/12/28 20:48:22-08:00 akpm@bix.(none) +0 -0
#   Auto merged
# 
# Documentation/kernel-parameters.txt
#   2004/12/28 20:48:22-08:00 akpm@bix.(none) +0 -0
#   Auto merged
# 
# ChangeSet
#   2004/12/25 08:59:48-08:00 akpm@bix.(none) 
#   Merge bix.(none):/usr/src/bk25 into bix.(none):/usr/src/bk-input
# 
# drivers/input/tsdev.c
#   2004/12/25 08:59:44-08:00 akpm@bix.(none) +0 -5
#   Auto merged
# 
# drivers/input/mousedev.c
#   2004/12/25 08:59:44-08:00 akpm@bix.(none) +0 -2
#   Auto merged
# 
# drivers/input/joydev.c
#   2004/12/25 08:59:44-08:00 akpm@bix.(none) +0 -2
#   Auto merged
# 
# drivers/input/evdev.c
#   2004/12/25 08:59:44-08:00 akpm@bix.(none) +0 -2
#   Auto merged
# 
# ChangeSet
#   2004/12/16 13:33:00-08:00 akpm@bix.(none) 
#   Merge bk://kernel.bkbits.net/vojtech/input
#   into bix.(none):/usr/src/bk-input
# 
# drivers/usb/input/hid-core.c
#   2004/12/16 13:32:57-08:00 akpm@bix.(none) +0 -0
#   Auto merged
# 
# Documentation/kernel-parameters.txt
#   2004/12/16 13:32:57-08:00 akpm@bix.(none) +0 -0
#   Auto merged
# 
diff -Nru a/drivers/input/keyboard/atkbd.c b/drivers/input/keyboard/atkbd.c
--- a/drivers/input/keyboard/atkbd.c	2005-01-13 16:57:34 -08:00
+++ b/drivers/input/keyboard/atkbd.c	2005-01-13 16:57:34 -08:00
@@ -756,6 +756,10 @@
 		set_bit(BTN_MIDDLE, atkbd->dev.keybit);
 	}
 
+	atkbd->dev.keycode = atkbd->keycode;
+	atkbd->dev.keycodesize = sizeof(unsigned char);
+	atkbd->dev.keycodemax = ARRAY_SIZE(atkbd_set2_keycode);
+
 	for (i = 0; i < 512; i++)
 		if (atkbd->keycode[i] && atkbd->keycode[i] < ATKBD_SPECIAL)
 			set_bit(atkbd->keycode[i], atkbd->dev.keybit);
@@ -803,10 +807,6 @@
 
 	if (atkbd->softrepeat)
 		atkbd->softraw = 1;
-
-	atkbd->dev.keycode = atkbd->keycode;
-	atkbd->dev.keycodesize = sizeof(unsigned char);
-	atkbd->dev.keycodemax = ARRAY_SIZE(atkbd_set2_keycode);
 
 	serio->private = atkbd;
 
diff -Nru a/drivers/input/mouse/logips2pp.c b/drivers/input/mouse/logips2pp.c
--- a/drivers/input/mouse/logips2pp.c	2005-01-13 16:57:33 -08:00
+++ b/drivers/input/mouse/logips2pp.c	2005-01-13 16:57:33 -08:00
@@ -245,7 +245,8 @@
  * Set up input device's properties based on the detected mouse model.
  */
 
-static void ps2pp_set_model_properties(struct psmouse *psmouse, struct ps2pp_info *model_info)
+static void ps2pp_set_model_properties(struct psmouse *psmouse, struct ps2pp_info *model_info,
+				       int using_ps2pp)
 {
 	if (model_info->features & PS2PP_SIDE_BTN)
 		set_bit(BTN_SIDE, psmouse->dev.keybit);
@@ -279,6 +280,16 @@
 		case PS2PP_KIND_TP3:
 			psmouse->name = "TouchPad 3";
 			break;
+
+		default:
+			/*
+			 * Set name to "Mouse" only when using PS2++,
+			 * otherwise let other protocols define suitable
+			 * name
+			 */
+			if (using_ps2pp)
+				psmouse->name = "Mouse";
+			break;
 	}
 }
 
@@ -371,7 +382,7 @@
 			clear_bit(BTN_RIGHT, psmouse->dev.keybit);
 
 		if (model_info)
-			ps2pp_set_model_properties(psmouse, model_info);
+			ps2pp_set_model_properties(psmouse, model_info, use_ps2pp);
 	}
 
 	return use_ps2pp ? 0 : -1;
